csharp/ql/src/Security Features/CWE-798/HardcodedCredentials.ql

```diff
@@ -11,13 +11,21 @@
  *       external/cwe/cwe-798
  */
 import csharp
-private import semmle.code.csharp.security.dataflow.HardcodedCredentials::HardcodedCredentials
+import semmle.code.csharp.security.dataflow.HardcodedCredentials::HardcodedCredentials
+import semmle.code.csharp.dataflow.DataFlow::DataFlow::PathGraph
 
-from TaintTrackingConfiguration c, Source source, Sink sink, string value
-where c.hasFlow(source, sink)
+from TaintTrackingConfiguration c, Source source, Sink sink, DataFlow::PathNode sourcePath, DataFlow::PathNode sinkPath,
+  string value
+where
+  source = sourcePath.getNode() and
+  sink = sinkPath.getNode() and
+  c.hasFlow(source, sink) and
```

Comment:
use `c.hasFlowPath(sourcePath, sinkPath)` instead.

Reply:
No, in this case it gives different results due to hackery around sources and sinks. The configuration's `hasFlow` has been changed, and `hasFlow` isn't `final` so I guess it's allowed.